lib/client.js

```diff
@@ -45,8 +46,89 @@ function Client (broker, conn, req) {
 
   this.disconnected = false
 
+  this.ipAddress = null
+  this.ipFamily = null
+  this.isWebsocket = false
+  this.isProxied = false
+
   this.parser.on('packet', enqueue)
 
+  function protocolDecoder (data) {
```

Comment:
Can you please move this function out of the constructor? It will cost more memory to execute.

Reply:
It is now in an external module -> protocol-decoder.js.